integration/airflow/openlineage/airflow/extractors/bigquery_check_extractors.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import traceback
+
+from abc import abstractmethod
+from typing import List, Optional
+
+from openlineage.airflow.extractors.base import TaskMetadata
+from openlineage.airflow.extractors.bigquery_extractor import BigQueryExtractor
+from openlineage.airflow.utils import get_job_name
+from openlineage.client.facet import SqlJobFacet
+from openlineage.common.provider.bigquery import BigQueryDatasetsProvider, BigQueryErrorRunFacet
+from facet_builders import (
+    build_check_facets,
+    build_value_check_facets,
+    build_interval_check_facets,
+    build_threshold_check_facets
+)
+
+logger = logging.getLogger(__name__)
+
+
+class BaseBigQueryCheckExtractor(BigQueryExtractor):
+    default_schema = 'public'
+
+    def __init__(self, operator):
+        super().__init__(operator)
+
+    def extract_on_complete(self, task_instance) -> Optional[TaskMetadata]:
+        logger.debug(f"extract_on_complete({task_instance})")
+        context = self.parse_sql_context()
+
+        try:
+            bigquery_job_id = self._get_xcom_bigquery_job_id(task_instance)
```

Comment:
This could be deduplicated with BigQueryExtractor, especially that you're subclassing it